client/src/components/lobby/Lobby.js

```diff
@@ -28,26 +37,38 @@ const Lobby = (): React$Element<any> => {
     });
   }, [socket]);
 
-  const firstUpdate = useRef(true);
   useEffect(() => {
-    if (!firstUpdate.current) {
-      socket.on("user-joined", (username: string) => {
-        const userInLobby: UserInLobbyT = {
-          username: username,
-          state: "Waiting",
-        };
-        const updatedUserList: Array<UserInLobbyT> = [...userList, userInLobby];
-        setUserList(updatedUserList);
-      });
-      return () => {
-        socket.off("user-joined");
+    socket.on("user-joined", (username: string) => {
+      const userInLobby: UserInLobbyT = {
+        username: username,
+        state: "Waiting",
       };
-    }
-    firstUpdate.current = false;
+      const updatedUserList: Array<UserInLobbyT> = [...userList, userInLobby];
+      if(isMounted.current) setUserList(updatedUserList);
+    });
+    socket.on("user-is-ready", (userIsReady: UserIsReadyT) => {
+      const updatedUserList: Array<UserInLobbyT> =
+        userList.map(
+          (userInLobby: UserInLobbyT): UserInLobbyT => 
+            userInLobby.username === userIsReady.username
+            ? {
+              ...userInLobby,
+              writtenCharacter: userIsReady.writtenCharacter,
+              state: "Ready",
+            } 
+            : userInLobby
+        ).filter((userInLobby: UserInLobbyT) => userInLobby);
```

Comment:
Is this filter doing something?

Reply:
map by default returns ?any, so is only to verify that this map filter will return an `Array<UserInLobbyT>` and not null